Skip Return Invoices When Outside of NA.

braidn opened the pull request:
If an order has been placed outside of North America, we don't want to
issue a return invoice.

braidn committed:
Skip Return Invoices When Outside of NA.

If an order has been placed outside of North America, we don't want to
issue a return invoice.

bryanmtl commented on app/models/spree_avatax/return_invoice.rb:
```diff
@@ -31,7 +31,7 @@ class << self
     #   return invoice.
     def generate(reimbursement)
-      if !SpreeAvatax::Config.enabled
+      if !SpreeAvatax::Config.enabled || !reimbursement.order.store.country.iso =~ /(US|CA)/
```
Maybe make this a method on the order? `north_american_order?`. Can we rely on the `store_id` on the order as an alternative?

braidn replied:
We can't rely on the store being assigned to a specific country since country is not a validation on the store, however the store is a validation on an order. Do we want to define that order instance method here or in the host app? We could also use the store_id's. Those are highly unlikely to change.

braidn replied:
Actually... store_id's might be fine.

DanielWright replied:
Magic numbers are always going to be a smell due to brittleness, even if they're unlikely to change in our specific instance.

I think I might have checked whether the tax-adjustments on the original order were Avatax-calculated:

```
reimbursement.order.all_adjustments.tax.all?(&:avatax?)
```

braidn committed:
Logic Tweak To North American Order.

No longer rely on the country of a store but, work with hard coded store
values that can be compared directly on the order.

braidn committed:
Return If There Is No Sales Invoice.

If there is no sales invoice to the order, it doesn't make a whole lot
of sense to process a return invoice.
